godot-core/src/builtin/projection.rs

```diff
@@ -0,0 +1,443 @@
+    /// plane.
+    ///
+    /// _Godot equivalent: Projection.get_projection_plane()_
+    pub fn projection_plane(&self, plane: ProjectionPlane) -> Plane {
+        self.as_inner().get_projection_plane(plane as i64)
+    }
+
+    /// Returns the dimensions of the viewport plane that this Projection
+    /// projects positions onto, divided by two.
+    ///
+    /// _Godot equivalent: Projection.get_viewport_half_extents()_
+    pub fn viewport_half_extents(&self) -> Vector2 {
+        self.as_inner().get_viewport_half_extents()
+    }
+
+    /// Returns the distance for this Projection beyond which positions are
+    /// clipped.
+    ///
+    /// _Godot equivalent: Projection.get_z_far()_
+    pub fn z_far(&self) -> f64 {
+        self.as_inner().get_z_far()
+    }
+
+    /// Returns the distance for this Projection before which positions are
+    /// clipped.
+    ///
+    /// _Godot equivalent: Projection.get_z_near()_
+    pub fn z_near(&self) -> f64 {
+        self.as_inner().get_z_near()
+    }
+
+    /// Returns a Projection that performs the inverse of this Projection's
+    /// projective transformation.
+    ///
+    /// _Godot equivalent: Projection.inverse()_
+    pub fn inverse(self) -> Self {
+        self.glam(|mat| mat.inverse())
+    }
+
+    /// Returns `true` if this Projection performs an orthogonal projection.
+    ///
+    /// _Godot equivalent: Projection.is_orthogonal()_
+    pub fn is_orthogonal(&self) -> bool {
+        self.as_inner().is_orthogonal()
+    }
+
+    /// Returns a Projection with the X and Y values from the given [`Vector2`]
+    /// added to the first and second values of the final column respectively.
+    ///
+    /// _Godot equivalent: Projection.jitter_offseted()_
+    pub fn jitter_offseted(&self, offset: Vector2) -> Self {
+        self.as_inner().jitter_offseted(offset)
+    }
+
+    /// Returns a Projection with the near clipping distance adjusted to be
+    /// `new_znear`.
+    ///
+    /// Note: The original Projection must be a perspective projection.
+    ///
+    /// _Godot equivalent: Projection.perspective_znear_adjusted()_
+    pub fn perspective_znear_adjusted(&self, new_znear: f64) -> Self {
+        self.as_inner().perspective_znear_adjusted(new_znear)
+    }
+
+    #[doc(hidden)]
+    pub(crate) fn as_inner(&self) -> InnerProjection {
+        InnerProjection::from_outer(self)
+    }
+}
+
+impl From<Transform3D> for Projection {
+    fn from(trans: Transform3D) -> Self {
+        trans.glam(Mat4::from)
+    }
+}
+
+impl Default for Projection {
+    fn default() -> Self {
+        Self::IDENTITY
+    }
+}
+
+impl Mul for Projection {
+    type Output = Self;
+
+    fn mul(self, rhs: Self) -> Self::Output {
+        self.glam2(&rhs, |a, b| a * b)
+    }
+}
+
+impl Mul<Vector4> for Projection {
+    type Output = Vector4;
+
+    fn mul(self, rhs: Vector4) -> Self::Output {
+        self.glam2(&rhs, |m, v| m * v)
+    }
+}
+
+impl GlamType for Mat4 {
```

Comment:
Could we keep glam types fully-qualified, i.e. `glam::Mat4`? This helps recognizability of foreign types, and there are not too many places where we need to name them, thanks to the convenience methods.